apps/web/src/components/Activity/ActivityItem.tsx

```diff
@@ -0,0 +1,105 @@
+import { useIntl } from "react-intl";
+import { tv } from "tailwind-variants";
+
+import {
+  FragmentType,
+  getFragment,
+  graphql,
+  Maybe,
+  Scalars,
+} from "@gc-digital-talent/graphql";
+import { commonMessages } from "@gc-digital-talent/i18n";
+import { formatDate, parseDateTimeUtc } from "@gc-digital-talent/date-helpers";
+
+import { getFullNameLabel } from "~/utils/nameUtils";
+
+import {
+  getEventInfo,
+  icon,
+  JSONRecord,
+  normalizePropKeys,
+  updatedAfterPublish,
+} from "./utils";
+
+const activityItem = tv({
+  base: "flex flex-col justify-between gap-6 py-6 sm:flex-row",
+  variants: {
+    border: {
+      true: "border-t-2 border-gray-200 dark:border-gray-500",
+      false: "",
+    },
+  },
+});
+
+const ActivityItem_Fragment = graphql(/** GraphQL */ `
+  fragment ActivityItem on Activity {
+    id
+    causer {
+      id
+      firstName
+      lastName
+    }
+    properties
+    event
+    createdAt
+  }
+`);
+
+interface ActivityItemProps {
+  query: FragmentType<typeof ActivityItem_Fragment>;
+  className?: string;
+  border?: boolean;
+  publishedAt?: Maybe<Scalars["DateTime"]["output"]>;
+}
+
+const ActivityItem = ({
+  query,
+  className,
+  border,
+  publishedAt,
+}: ActivityItemProps) => {
+  const intl = useIntl();
+  const item = getFragment(ActivityItem_Fragment, query);
+  const isAfterPublish = updatedAfterPublish(item.createdAt, publishedAt);
+  const propsObj = JSON.parse(String(item.properties)) as JSONRecord;
```

Comment:
This cast makes me a little nervous.  Could we test that those properties actually exist on the JSON object first?

Reply:
so, we will need the cast regarldess but I think it should be much safer after the changes made in [refactor to improve properties type](https://github.com/GCTC-NTGC/gc-digital-talent/pull/15120/commits/02de3d2b3e04eaadcbfe5d110c0116d3ebae625d)